drivers/clk/clk-axi-clkgen.c

```
@@ -574,12 +575,6 @@ static int axi_clkgen_probe(struct platform_device *pdev)
 	if (IS_ERR(axi_clkgen->base))
 		return PTR_ERR(axi_clkgen->base);
 
-	axi_clkgen_read(axi_clkgen, ADI_AXI_REG_VERSION,
```

this isn't an issue yet, but would it be a problem to leave the `axi_clkgen_read()` here? and add the axi_clk setup before this part?

it would also mean that `clk_disable_unprepare(axi_clkgen->axi_clk);` would go after disabling the parent-clocks [during remove & the error path], but that would actually make some sense, since [in some potential future versions of this driver] someone would want to squeeze some register reads/writes before

---

I'm not sure I understand this comment, is this on the latest revision I sent?
On the diff I see, the axi_clkgen_read is right after the `s_axi_aclk` setup.

---

the suggestion is to add here, this block:
```
	axi_clkgen->axi_clk = devm_clk_get(&pdev->dev, "s_axi_aclk");
	if (IS_ERR(axi_clkgen->axi_clk)) {
		dev_err(&pdev->dev, "failed to get s_axi_aclk\n");
		return PTR_ERR(axi_clkgen->axi_clk);
	}
```

naturally `clk_disable_unprepare(axi_clkgen->axi_clk);` would need to be done after the parent-clocks disable-unprepare;
and the error-paths of the parent-clocks init would need to be updated;

this change can be done now, or later; either is fine from my side;
it's the only remaining comment i have for this set;

---

Hi @commodo 

I'm sorry if I'm still confused but in the latest revision I sent, the `devm_clk_get(&pdev->dev, "s_axi_aclk")` and the `axi_clkgen_read` were moved after the clock setup code (https://github.com/Liambeguin/linux-1/blob/axi-clkgen/drivers/clk/clk-axi-clkgen.c#L615).

So right now, the first `axi_clkgen_read` happens after the clock setup. Is that what you meant?

---

i'm saying that we could check for the axi_clk first and fail early [and not enable any parent clocks] if the axi_clk is not ready/available;

---

Oh! sorry for not getting that earlier... I'll update and resend